checker/src/main/java/org/checkerframework/checker/initialization/InitializationFieldAccessAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,200 @@
+package org.checkerframework.checker.initialization;
+
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.IdentifierTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.Tree;
+
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.dataflow.analysis.AnalysisResult;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.framework.type.GenericAnnotatedTypeFactory;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.TreeUtils;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.type.TypeMirror;
+
+/** The type factory for the {@link InitializationFieldAccessSubchecker}. */
+public class InitializationFieldAccessAnnotatedTypeFactory
+        extends InitializationParentAnnotatedTypeFactory {
+
+    /**
+     * Create a new InitializationFieldAccessAnnotatedTypeFactory.
+     *
+     * @param checker the checker to which the new type factory belongs
+     */
+    public InitializationFieldAccessAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(checker);
+        postInit();
+    }
+
+    @Override
+    protected InitializationAnalysis createFlowAnalysis() {
+        return new InitializationAnalysis(checker, this);
+    }
+
+    @Override
+    protected void performFlowAnalysis(ClassTree classTree) {
+        // Only perform the analysis if initialization checking is turned on.
+        if (!checker.hasOption("assumeInitialized")) {
+            super.performFlowAnalysis(classTree);
+            ;
+        }
+    }
+
+    /**
+     * Returns the flow analysis.
+     *
+     * @return the flow analysis
+     * @see #getFlowResult()
+     */
+    InitializationAnalysis getAnalysis() {
```

Comment:
Can you explicitly add `/*package-private*/` throughout the PR if that is the intended visibility?